Make Habitica setup more user-friendly

tr4nt0r opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

The Config Flow for setting up the Habitica integration is now a lot more user-friendly. Users can now user their normal username/e-mail and password combination they also use to login to Habitica, instead having to handle with their User ID and API Token. The Config Flow then stores the User ID and API Token which are returned in the JSON response of the login endpoint. 

Because Habitica is open source, users could want to connect to a different instance than the official Habitica, even to a self-hosted one. Therefore users can choose between "Login to Habitica" and an "Advanced Configuration". The Advanced Configuration still offers the same options as before, like using User ID and API Token and the URL to point to a different instance. Just an option to disable SSL verification was added, in case it is an instance with a self-signed certificate.

May also need a Documentation PR.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

joostlek commented on homeassistant/components/habitica/config_flow.py:
```diff
@@ -2,83 +2,167 @@
-        errors = {}
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["login", "advanced"],
```
But both are technically logging in, so can we think of better names?

joostlek replied:
habitica and self_hosted would do I think

tr4nt0r replied:
Legacy Login xD

joostlek replied:
normies and the rest

tr4nt0r replied:
Maybe "Login to Habitica" and "Login to other instances" ??

joostlek replied:
sounds good

joostlek commented on homeassistant/components/habitica/__init__.py:
```diff
@@ -141,19 +142,20 @@ async def handle_api_call(call: ServiceCall) -> None:
     api = HAHabitipyAsync(config_dict)
     user = await api.user.get()
-    if name is None:
-        name = user["profile"]["name"]
+    if name != user["auth"]["local"]["username"]:
         hass.config_entries.async_update_entry(
             entry,
-            data={**entry.data, CONF_NAME: name},
+            data={**entry.data, CONF_NAME: user["auth"]["local"]["username"]},
         )
```
Honestly, we should get rid of saving the name in the entry data

tr4nt0r replied:
yeah, the service somehow still needs it, otherwise it is not used anywhere

tr4nt0r replied:
i got rid of the username 🎉

tr4nt0r replied:
not possible until we got rid of the api service, so leaving it for now

joostlek commented on homeassistant/components/habitica/config_flow.py:
```diff
@@ -2,83 +2,167 @@
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(user_input[CONF_API_USER])
```
If I have 2 habitica servers with the same username, this doesn't work. Should we include the url?

tr4nt0r replied:
the User ID is a ramdomly generated UUID, so the chances for a collision should be pretty low

joostlek commented on homeassistant/components/habitica/strings.json:
```diff
@@ -4,18 +4,32 @@
-          "api_user": "Habitica’s API user ID",
-          "api_key": "[%key:common::config_flow::data::api_key%]"
+          "api_user": "User ID",
+          "api_key": "API Token",
```
What does habitica use to name their API token/key?

tr4nt0r replied:
they name it `User ID` and  `API Token`

joostlek commented on tests/components/habitica/test_config_flow.py:
```diff
@@ -3,27 +3,61 @@
-async def test_form(hass: HomeAssistant) -> None:
-    """Test we get the form."""
 
+async def test_user_menu(hass: HomeAssistant) -> None:
+    """Test if we get the menu selection."""
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
+
+    assert result["type"] is FlowResultType.MENU
+    assert result["menu_options"] == ["login", "advanced"]
+    assert result["step_id"] == "user"
+
+
+async def test_form_login(hass: HomeAssistant) -> None:
+    """Test we get the login form."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "login"}
```
Just combine these tests

tr4nt0r replied:
hope thats better now

joostlek commented on tests/components/habitica/test_config_flow.py:
```diff
@@ -37,73 +71,123 @@ async def test_form(hass: HomeAssistant) -> None:
+    ],
+)
+async def test_form_login_errors(hass: HomeAssistant, raise_error, text_error) -> None:
     """Test we handle invalid credentials error."""
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
+        DOMAIN, context={"source": "login"}
     )
```
Let's start the flow like normal

autinerd committed:
Ignore Ruff SIM103 (#115732)

Co-authored-by: J. Nick Koston <nick@koston.org>

puddly committed:
Add a missing `addon_name` placeholder to the SkyConnect config flow (#117089)

bdraco committed:
Fix config entry _async_process_on_unload being called for forwarded platforms (#117084)

mib1185 committed:
Store runtime data inside the config entry in Tuya (#116822)

mletenay committed:
Goodwe Increase max value of export limit to 200% (#117090)

Lash-L committed:
Add test data for Zeo and Dyad devices to Roborock (#117054)

mletenay committed:
Bump goodwe to 0.3.5 (#117115)

Lash-L committed:
Add better testing to vacuum platform (#112523)

* Add better testing to vacuum platform

* remove state strings

* some of the MR comments

* move MockVacuum

* remove manifest extra

* fix linting

* fix other linting

* Fix create entity calls

* Format

* remove create_entity

* change to match notify

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cdce8p committed:
Use HassKey for importlib helper (#117116)

bdraco committed:
Move available property in BasePassiveBluetoothCoordinator to PassiveBluetoothDataUpdateCoordinator (#117056)

dependabot[bot] committed:
Bump github/codeql-action from 3.25.3 to 3.25.4 (#117127)

joostlek committed:
Add airgradient integration (#114113)

joostlek committed:
Add base entity for Airgradient (#117135)

dieselrabbit committed:
Add screenlogic service tests (#116356)

bdraco committed:
Speed up test teardown when no config entries are loaded (#117095)

Avoid the gather call when there are no loaded config entries

jbouwh committed:
Fix typo in xiaomi_ble translation strings (#117144)

Thomas55555 committed:
Catch auth exception in husqvarna automower (#115365)

* Catch AuthException in Husqvarna Automower

* don't use getattr

* raise ConfigEntryAuthFailed

autinerd committed:
Bump ruff to 0.4.4 (#117154)

bdraco committed:
Handle tilt position being None in HKC (#117141)

tizianodeg committed:
Raise ServiceValidationError in Nibe climate services (#117171)

Fix ClimateService to rise ServiceValidationError for stack free logs

RJPoelstra committed:
Improve base entity state in Vogel's MotionMount integration (#109043)

* Update device info when name changes

* Entities now report themselves as being unavailable when the MotionMount is disconnected

* Don't update device_info when name changes

* Use `device_entry` property to update device name

* Assert device is available

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Add missing import

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

CoRfr committed:
Add standard deviation calculation to group (#112076)

* Add standard deviation calculation to group

* Add missing bits

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

Bre77 committed:
Refactor Teslemetry integration (#112480)

* Refactor Teslemetry

* Add abstractmethod

* Remove unused timestamp const

* Ruff

* Fix

* Update snapshots

* ruff

* Ruff

* ruff

* Lint

* Fix tests

* Fix tests and diag

* Refix snapshot

* Ruff

* Fix

* Fix bad merge

* has as property

* Remove _handle_coordinator_update

* Test and error changes

jbouwh committed:
Migrate file integration to config entry (#116861)

* File integration entry setup

* Import to entry and tests

* Add config flow

* Exception handling and tests

* Add config flow tests

* Add issue for micration and deprecation

* Check whole entry data for uniqueness

* Revert changes change new notify entity

* Follow up on code review

* Keep name service option

* Also keep sensor name

* Make name unique

* Follow up comment

* No default timestamp needed

* Remove default name as it is already set

* Use links

Bre77 committed:
Add energy site coordinator to Teslemetry (#117184)

* Add energy site coordinator

* Add missing string

* Add another missing string

* Aprettier

Links2004 committed:
Add ESPhome discovery via MQTT (#116499)

Co-authored-by: J. Nick Koston <nick@koston.org>

edenhaus committed:
Bump deebot-client to 7.2.0 (#117189)

cdce8p committed:
Import TypedDict from typing (#117161)

jbouwh committed:
Log some mqtt of the discovery logging at debug level (#117185)

Shulyaka committed:
Update media_player intent schema (#116793)

Update media_player/intent.py

dgomes committed:
Bump pytrydan to 0.6.0 (#117162)

Shulyaka committed:
Update SetPositionIntentHandler intent schema (#116794)

Update SetPositionIntentHandler

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

marcolivierarsenault committed:
Add update_without_throttle to ecobee number (#116504)

add update_without_throttle

jbouwh committed:
Fix File entry setup config parsing whole YAML config (#117206)

Fix File entry setup config parsingwhole YAML config

bdraco committed:
Add state check to config entry setup to ensure it cannot be setup twice (#117193)

bdraco committed:
Ensure config entry setup lock is held when removing a config entry (#117086)

bdraco committed:
Fix flakey sonos test teardown (#117222)

https://github.com/home-assistant/core/actions/runs/9039805087/job/24843300480?pr=117214

allenporter committed:
Bump ical to 8.0.1 (#117219)

bdraco committed:
Use async_get_loaded_integration in config_entries (#117192)

jjlawren committed:
Bump SoCo to 0.30.4 (#117212)

allenporter committed:
Bump pyrainbird to 6.0.1 (#117217)

* Bump pyrainbird to 6.0.0

* Bump to 6.0.1

bdraco committed:
Fix flakey advantage_air test (#117224)

bdraco committed:
Only load translations for an integration once per test session (#117118)

gjohansson-ST committed:
Add device class to Command Line cover (#117183)

systemcrash committed:
Prevent shutdown fault-log trace-back (#116735)

Closes issue #116710

bdraco committed:
Speed up init and finish flow (#117226)

Since every flow now has to check for single config entry, change
the check to see if a config entry exists first before calling
the _support_single_config_entry_only since _support_single_config_entry_only
has to load the integration which adds up quite a bit in test runs

emontnemery committed:
Use MockConfigEntry in unifi tests (#117238)

emontnemery committed:
Rename some runner tests (#117249)

emontnemery committed:
Rename some translation helper tests (#117248)

emontnemery committed:
Rename some rflink tests (#117247)

emontnemery committed:
Rename some MQTT tests (#117246)

emontnemery committed:
Sort asserts in config config_entries tests (#117244)

raman325 committed:
Fix zwave_js discovery logic for node device class (#117232)

* Fix zwave_js discovery logic for node device class

* simplify check

hahn-th committed:
Bump homematicip to 1.1.1 (#117175)

EnjoyingM committed:
Bump wolf-comm to 0.0.8 (#117218)

teharris1 committed:
Bump pyinsteon to 1.6.1 (#117196)

* Bump pyinsteon

* Bump pyinsteon

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

raman325 committed:
Start using runtime_data for zwave_js (#117261)

* Start using runtime_data for zwave_js

* fix bug

emontnemery committed:
Use mock_config_flow helper in config tests (#117245)

emontnemery committed:
Use mock_config_flow helper in config_entries tests (#117241)

emontnemery committed:
Use mock_config_flow helper in bootstrap tests (#117240)

emontnemery committed:
Sort parameters to MockConfigEntry (#117239)

jbouwh committed:
Add shared notify service migration repair helper (#117213)

* Add shared notifiy service migration repair helper

* Delete ecobee repairs.py

* Update dependency

* Fix file test

* Fix homematic tests

* Improve tests for file  and homematic

joostlek committed:
Refactor V2C tests (#117264)

* Refactor V2C tests

* Refactor V2C tests

* Refactor V2C tests

* Refactor V2C tests

* Update tests/components/v2c/conftest.py

* Refactor V2C tests

emontnemery committed:
Tweak config_entries tests (#117242)

bdraco committed:
Ensure config entry operations are always holding the lock (#117214)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

abmantis committed:
Fix docstring in Idasen Desk (#117280)

bdraco committed:
Small speed up to setting up integrations and config entries (#117278)

* Small speed up to setting up integration and config entries

When profiling tests, I noticed many calls to get_running_loop. In the places
where we are already in a coro, pass the existing loop so it does not have to
be looked up. I did not do this for places were we are not in a coro since there
is risk that an integration could be doing a non-thread-safe call and its better
that the code raises when trying to fetch the running loop vs the performance
improvement for these cases.

* fix merge

* missed some

emontnemery committed:
Use `MockConfigEntry` in hue tests (#117237)

Use MockConfigEntry in hue tests

bdraco committed:
Use a dictcomp to reconstruct DeviceInfo in the device_registry (#117286)

Use a dictcomp to reconstruct DeviceInfo

a dictcomp is faster than many sets on the dict by at least
25%

We call this for nearly every device in the registry at
startup

scop committed:
Bump upcloud-api to 2.5.1 (#117231)

Upgrade upcloud-python-api to 2.5.1

- https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/v2.0.1
- https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/v2.5.0
- https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/v2.5.1

bdraco committed:
Fix mimetypes doing blocking I/O in the event loop (#117292)

The first time aiohttp calls mimetypes, it will load the mime.types file

We now init the db in the executor to avoid blocking the event loop

bdraco committed:
Small speed ups to bootstrap tests (#117285)

jbouwh committed:
Increase MQTT broker socket buffer size (#117267)

* Increase MQTT broker socket buffer size

* Revert unrelated change

* Try to increase buffer size

* Set INITIAL_SUBSCRIBE_COOLDOWN back to 0.5 sec

* Sinplify and add test

* comments

* comments

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

cdce8p committed:
Improve bluetooth generic typing (#117157)

jbouwh committed:
Spelling of controlling in mqtt valve tests (#117301)

bdraco committed:
Fix blocking I/O in the event loop to get MacOS system_info (#117290)

* Fix blocking I/O in the event look to get MacOS system_info

* split pr

* Update homeassistant/helpers/system_info.py

Co-authored-by: Jan-Philipp Benecke <jan-philipp@bnck.me>

* Update homeassistant/helpers/system_info.py

---------

Co-authored-by: Jan-Philipp Benecke <jan-philipp@bnck.me>

Bre77 committed:
Use entry runtime data in Teslemetry (#117283)

* runtime_data

* runtime_data

* Remove some code

* format

* Fix missing entry.runtime_data

bdraco committed:
Fix flume doing blocking I/O in the event loop (#117293)

constructing FlumeData opens files

jbouwh committed:
Fix import on File config entry and other improvements (#117210)

* Address comments

* Remove Name option for File based sensor

* Make sure platform schema is applied

jbouwh committed:
Fix typo in mqtt test name (#117305)

jbouwh committed:
Migrate Tibber notify service (#116893)

* Migrate tibber notify service

* Tests and repair flow

* Use notify repair flow helper

* Cleanup strings after using helper, use HomeAssistantError

* Add entry state assertions to unload test

* Update comment

* Update comment

cdce8p committed:
Remove ConfigEntry runtime_data on unload (#117312)

tkdrob committed:
Add test for radarr update failure (#116882)

joostlek committed:
Fix Aurora naming (#117314)

joostlek committed:
Clean up aurora (#117315)

* Clean up aurora

* Fix

* Fix

* Fix

bdraco committed:
Refactor asyncio loop protection to improve performance (#117295)

bdraco committed:
Enable open protection in the event loop (#117289)

bdraco committed:
Fix squeezebox blocking startup (#117331)

fixes #117079

tronikos committed:
Call Google Assistant SDK service using async_add_executor_job (#117325)

raman325 committed:
Bump zwave-js-server-python to 0.56.0 (#117288)

* Bump zwave-js-server-python to 0.56.0

* Fix deprecation warning

* Fix tests

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

mback2k committed:
Add camera recording service to blink (#110612)

Add camera clip recording service to blink

Revival of #46598 by @fronzbot, therefore:

Co-authored-by: Kevin Fronczak <kfronczak@gmail.com>

Bre77 committed:
Improve error handling in Teslemetry (#117336)

* Improvement command handle

* Add test for ignored reasons

balloob committed:
Add gh cli to dev container (#117321)

emontnemery committed:
Fix typo and useless default in config_entries (#117346)

bouwew committed:
Implement runtime data for Plugwise (#117172)

emontnemery committed:
Remove migration of config entry data pre version 0.73 (#117345)

emontnemery committed:
Alphabetize some parts of config_entries (#117347)

emontnemery committed:
Remove options from FlowResult (#117351)

emontnemery committed:
Add minor version + migration to config entry store (#117350)

jbouwh committed:
Migrate File notify entity platform (#117215)

* Migrate File notify entity platform

* Do not load legacy notify service for new  config entries

* Follow up comment

* mypy

* Correct typing

* Only use the name when importing notify services

* Make sure a name is set on new entires

laurence-presland committed:
Implement support for SwitchBot Meter, MeterPlus, and Outdoor Meter (#115522)

* Implement support for SwitchBot MeterPlus

Add temperature, humidity, and battery sensor entities for the MeterPlus device

* Rename GH username

* Update homeassistant/components/switchbot_cloud/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Refactor to use EntityDescriptions

Concat entity ID in SwitchBotCloudSensor init

* Remove __future__ import

* Make scan interval user configurable

* Add support for Meter and Outdoor Meter

* Revert "Make scan interval user configurable"

This reverts commit e256c35bb71e598cf879e05e1df21dff8456b09d.

* Remove device-specific default scan intervals

* Update homeassistant/components/switchbot_cloud/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/switchbot_cloud/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/switchbot_cloud/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fix ruff errors

* Reorder manifest keys

* Update CODEOWNERS

* Add sensor.py to coveragerc

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bieniu committed:
Support reconfigure flow in Nettigo Air Monitor integration  (#117318)

* Add reconfigure flow

* Fix input

* Add tests

---------

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

elupus committed:
Update to arcam 1.5.2 (#117375)

jbouwh committed:
Create helper for File config flow step handling (#117371)

bdraco committed:
Bump dbus-fast to 2.21.2 (#117195)

Thomas55555 committed:
Add reauth for missing token scope in Husqvarna Automower (#117098)

* Add repair for wrong token scope to Husqvarna Automower

* avoid new installations with missing scope

* tweaks

* just reauth

* texts

* Add link to correct account

* Update homeassistant/components/husqvarna_automower/strings.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/husqvarna_automower/strings.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/husqvarna_automower/strings.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Add comment

* directly assert mock_missing_scope_config_entry.state is loaded

* assert that a flow is started

* pass complete url to strings and simplify texts

* shorten long line

* address review

* simplify tests

* grammar

* remove obsolete fixture

* fix test

* Update tests/components/husqvarna_automower/test_init.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* test if reauth flow has started

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

wujiaqi committed:
Fix Lutron Serena Tilt Only Wood Blinds set tilt function (#117374)

bdraco committed:
Migrate restore_state to use the singleton helper (#117385)

bdraco committed:
Pass loop to create_eager_task in loops from more coros (#117390)

allenporter committed:
Import and cache supported feature enum flags only when needed (#117270)

* Import and cache supported feature enum flags only when needed

* Add comment aboud being loaded from executor.

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Use ConfigEntry runtime_data in Tailwind (#117404)

cdce8p committed:
Update uv to 0.1.43 (#117405)

frenck committed:
Update apprise to 1.8.0 (#117370)

cdce8p committed:
Update types packages (#117407)

epenet committed:
Use ConfigEntry runtime_data in advantage_air (#117408)

dependabot[bot] committed:
Bump github/codeql-action from 3.25.4 to 3.25.5 (#117409)

frenck committed:
Update gotailwind to 0.2.3 (#117402)

epenet committed:
Use ConfigEntry runtime_data in aemet (#117411)

epenet committed:
Use ConfigEntry runtime_data in agent_dvr (#117412)

cdce8p committed:
Update pylint to 3.1.1 (#117416)

epenet committed:
Move abode base entities to separate module (#117417)

FedDam committed:
Add climate platform to microBees (#111152)

* Add climate platform to microBees

* add list comprehension

* fixes

* fixes

* fixes

* fix multiline ternary

* use a generator expression instead of filter + lambda.

* bug fixes

* Update homeassistant/components/microbees/climate.py

---------

Co-authored-by: Marco Lettieri <m.lettieri@microbees.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

HipsterBrown committed:
Add Viam image processing integration (#101786)

* feat: scaffold integration, configure client

* feat: register services, allow API key auth flow

* feat: register detection, classification services

* test(viam): add test coverage

* chore(viam): update viam-sdk version

* fix(viam): add service schemas and translation keys

* test(viam): update config flow to use new selector values

* chore(viam): update viam-sdk to 0.11.0

* feat: add exceptions translation stings

* refactor(viam): use constant keys, defer filesystem IO execution

* fix(viam): add missing constants, resolve correct image for services

* fix(viam): use lokalize string refs, resolve more constant strings

* fix(viam): move service registration to async_setup

* refactor: abstract services into separate module outside of manager

* refactor(viam): extend common vol schemas

* refactor(viam): consolidate common service values

* refactor(viam): replace FlowResult with ConfigFlowResult

* chore(viam): add icons.json for services

* refactor(viam): use org API key to connect to robot

* fix(viam): close app client if user abort config flow

* refactor(viam): run ruff formatter

* test(viam): confirm 100% coverage of config_flow

* refactor(viam): simplify manager, clean up config flow methods

* refactor(viam): split auth step into auth_api & auth_location

* refactor(viam): remove use of SelectOptionDict for auth choice, update strings

* fix(viam): use sentence case for translation strings

* test(viam): create mock_viam_client fixture for reusable mock

bdraco committed:
Speed up loading YAML (#117388)

Co-authored-by: Dave T <17680170+davet2001@users.noreply.github.com>

cdce8p committed:
Update pytest warnings filter (#117413)

farmio committed:
Allow templates for enabling automation triggers (#114458)

* Allow templates for enabling automation triggers

* Test exception for non-limited template

* Use `cv.template` instead of `cv.template_complex`

* skip trigger with invalid enable template

instead of returning and thus not evaluating other triggers

farmio committed:
Add `knx.telegram` integration specific trigger; update KNX Interface device trigger (#107592)

* Add `knx.telegram` integration specific trigger

* Move implementation to trigger.py, use it from device_trigger

* test device_trigger

* test trigger.py

* Add "incoming" and "outgoing" and handle legacy device triggers

* work with mixed group address styles

* improve coverage

* Add no-op option

* apply changed linting rules

* Don't distinguish legacy device triggers from new ones

that's now supported since frontend has fixed default values of extra_fields

* review suggestion: reuse trigger schema for device trigger extra fields

* cleanup for readability

* Remove no-op option

epenet committed:
Rename sharkiq coordinator module (#117429)

joostlek committed:
Use snapshot platform helper in Flexit bacnet (#117428)

epenet committed:
Move esphome coordinator to separate module (#117427)

epenet committed:
Move abode service registration (#117418)

bdraco committed:
Adjust thread safety check messages to point to developer docs (#117392)

dgomes committed:
Add diagnostic platform to utility_meter (#114967)

* Add diagnostic to identify next_reset

* Add test

* add next_reset attr

* Trigger CI

* set as _unrecorded_attributes

tkdrob committed:
Bump pyefergy to 22.5.0 (#117395)

Shulyaka committed:
Add context to `telegram_bot` events (#109920)

* Add context for received messages events

* Add context for sent messages events

* ruff

* ruff

* ruff

* Removed user_id mapping

* Add tests

epenet committed:
Move evil_genius_labs coordinator to separate module (#117435)

CFenner committed:
Add ability to change heating programs for heat pumps in ViCare integration (#110924)

* heating programs

* fix heating program

* fix heating program

* remove commented code

* simplify

* Update types.py

* update vicare_programs in init

epenet committed:
Rename flo coordinator module (#117438)

frenck committed:
Update wled to 0.17.1 (#117444)

cereal2nd committed:
Bump pyduotecno to 2024.5.0 (#117446)

ethemcemozkan committed:
Use integration fallback configuration for tado water heater fallback  (#111014)

* 103619 tado water heater fallback

* extracted a method to remove code duplication

* test cases and suggested changes

* tests

* util method for connector

* Update homeassistant/components/tado/climate.py

Co-authored-by: Andriy Kushnir <me@orhideous.name>

* missing import after applies suggestion

* early return

* simplify if statements

* simplify pr

* pr requested changes

* better docstring

---------

Co-authored-by: Andriy Kushnir <me@orhideous.name>

zweckj committed:
Re-introduce webhook to tedee integration (#110247)

* bring webhook over to new branch

* change log levels

* Update homeassistant/components/tedee/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* fix minor version

* ruff

* mock config entry version

* fix

* ruff

* add cleanup during webhook registration

* feedback

* ruff

* Update __init__.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update homeassistant/components/tedee/__init__.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* add downgrade test

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

andrewsayre committed:
Update SmartThings codeowners (#117448)

epenet committed:
Rename wemo coordinator module (#117437)

synesthesiam committed:
Add Assist timers (#117199)

* First pass at timers

* Move to separate file

* Refactor to using events

* Add pause/unpause/status

* Add ordinal

* Add test for timed Assist command

* Fix name matching

* Fix IntentHandleError

* Fix again

* Refactor to callbacks

* is_paused -> is_active

* Rename "set timer" to "start timer"

* Move tasks to timer manager

* More fixes

* Remove assist command

* Remove cancel by ordinal

* More tests

* Remove async on callbacks

* Export async_register_timer_handler

joostlek committed:
Refactor Linear tests (#116336)

synesthesiam committed:
Pass device_id to intent handlers (#117442)

mk-81 committed:
Fix Kodi on/off status (#117436)

* Fix Kodi Issue 104603

Fixes issue, that Kodi media player is displayed as online even when offline. The issue occurrs when using HTTP(S) only (no web Socket) integration after kodi was found online once.
Issue: In async_update the connection exceptions from self._kodi.get_players are not catched and therefore self._players (and the like) are not reset. The call of self._connection.connected returns always true for HTTP(S) connections.

Solution: Catch Exceptions from self._kodi.get_players und reset state in case of HTTP(S) only connection. Otherwise keep current behaviour.

* Fix Kodi Issue 104603 / code style adjustments

as requested

frenck committed:
Some minor cleanups in WLED (#117453)

Kane610 committed:
Make UniFi use runtime data (#117457)

rikroe committed:
Enable raising ConfigEntryAuthFailed on BMW coordinator init (#116643)

Co-authored-by: Richard <rikroe@users.noreply.github.com>

emontnemery committed:
Fix lying docstring in entity_platform (#117450)

kamiyo committed:
Add PM10 sensor to AirNow (#117432)

c0mputerguru committed:
Bump opower to 0.4.5 and use new account.id (#117330)

benvm committed:
Bump youless_api to  1.1.1 (#117459)

synesthesiam committed:
Remove "device_id" slot from timers (#117460)

Remove "device_id" slot

benvm committed:
Add water meter to Youless intergration (#117452)

Co-authored-by: Franck Nijhof <git@frenck.dev>

davet2001 committed:
Set integration type for aurora_abb_powerone (#117462)

Floris272 committed:
Separate Blue Current timestamp sensors (#111942)

cottsay committed:
Fix brand ID for Rainforest Automation (#113770)

mawoka-myblock committed:
Add integration for APsystems EZ1 microinverter (#114531)

* Add APsystems local API integration

* Fix session usage in config_flow in apsystems local api

* Remove skip check option for apsystems_loca api

* Update APsystems API dependency and increased test coverage to 100%

* Utilize EntityDescriptions for APsystems Local integration

* Ensure coverage entries are sorted (#114424)

* Ensure coverage entries are sorted

* Use autofix

* Adjust

* Add comment to coverage file

* test CI

* revert CI test

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Use patch instead of Http Mocks for APsystems API tests

* Fix linter waring for apsystemsapi

* Fix apsystemsapi test

* Fix CODEOWNERS for apsystemsapi

* Address small PR review changes for apsystems_local

* Remove wrong lines in coveragerc

* Add serial number for apsystems_local

* Remove option of custom refresh interval fro apsystems_local

* Remove function override and fix stale comments

* Use native device id and name storage instead of custom one for apsystems_local

* Use runtime_data for apsystems_local

* Don't store entry data in runtime data

* Move from apsystems_local to apsystems domain

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Fix intent_type type (#117469)

bdraco committed:
Improve thread safety check messages to better convey impact (#117467)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Add websocket API to get list of recorded entities (#92640)

* Add API to get list of recorded entities

* update for latest codebase

* ruff

* Update homeassistant/components/recorder/websocket_api.py

* Update homeassistant/components/recorder/websocket_api.py

* Update homeassistant/components/recorder/websocket_api.py

* add suggested test

dependabot[bot] committed:
Bump codecov/codecov-action from 4.3.1 to 4.4.0 (#117472)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

epenet committed:
Split homeassistant_alerts constants and coordinator (#117475)

epenet committed:
Rename gree coordinator module (#117474)

Sibgatulin committed:
Add options flow to the airq integration (#109337)

* Add support for options to airq integration

Expose to the user the following configuration:
1. A choice between fetching from the device either:
    a. the averaged (previous and the new default behaviour) or
    b. noisy momentary sensor reading
2. A toggle to clip (spuriously) negative sensor values (default
    functionality, previously unexposed)

To those ends:
- Introduce an `OptionsFlowHandler` alongside with a listener
  `AirQCoordinator.async_set_options`
- Introduce constants to handle represent options
- Add tests and strings

* Drop OptionsFlowHandler in favour of SchemaOptionsFlowHandler

Modify `AirQCoordinator.__init__` to accommodate the change in option
handling, and drop `async_set_options` which slipped through the
previous commit.

* Ruff formatting

emontnemery committed:
Revert "Add Viam image processing integration" (#117477)

frenck committed:
Revert "Bump opower to 0.4.5 and use new account.id" (#117476)

joostlek committed:
Rename add entities function in Aurora (#117480)

epenet committed:
Move ialarm coordinator to separate module (#117478)

davet2001 committed:
Set integration type for generic (#117464)

timmo001 committed:
Add exception translations to System Bridge integration (#112206)

* Add exception translations to System Bridge integration

* Add translated error to coordinator

* Refactor strings.json in system_bridge component

* Sort

* Add HomeAssistantError import

epenet committed:
Move gios coordinator to separate module (#117471)

epenet committed:
Move idasen_desk coordinator to separate module (#117485)

epenet committed:
Remove ignore-wrong-coordinator-module in pylint CI (#117479)

epenet committed:
Rename mikrotik coordinator module (#117488)

starkillerOG committed:
Reolink fix not unregistering webhook during ReAuth (#117490)

bdraco committed:
Fix non-thread-safe state write in tellduslive (#117487)

epenet committed:
Move melnor coordinator to separate module (#117486)

emontnemery committed:
Fix API creation for passwordless pi_hole (#117494)

epenet committed:
Move prusalink coordinators to separate module (#117495)

epenet committed:
Move mill coordinator to separate module (#117493)

amura11 committed:
Fix Fully Kiosk set config service (#112840)

* Fixed a bug that prevented setting Fully Kiosk config values using a template

* Added test to cover change

* Fixed issue identified by Ruff

* Update services.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cereal2nd committed:
Mark Duotecno entities unavailable when tcp goes down (#114325)

When the tcp connection to the duotecno smartbox goes down, mark all entities as unavailable.

epenet committed:
Move venstar coordinator to separate module (#117500)

marcelveldt committed:
Fix ghost events for Hue remotes  (#113047)

* Use report values for events

* adjust tests

tronikos committed:
Android TV Remote: Support launching any app by its application ID/package name (#116906)

Bumps androidtvremote2 to 0.1.1

d-ylee committed:
Add Jellyfin audio_codec optionflow (#113036)

* Fix #92419; Add Jellyfin audio_codec optionflow

* Use CONF_AUDIO_CODEC constant, clean up code based on suggestions

* Fixed typos

* Parameterize Tests

* Use parameterized test for jellyfin test media resolve

* Apply suggestions from code review

* Update homeassistant/components/jellyfin/config_flow.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

starkillerOG committed:
Bump reolink-aio to 0.8.10 (#117501)

AnilDaoud committed:
Improve error handing in kaiterra data retrieval when no aqi data is present (#112885)

* Update api_data.py

change log level on typeerror on line 103 from error to debug, it occurs too often to be useful as an error

* Update api_data.py

restore error level and add a type check instead

* Update homeassistant/components/kaiterra/api_data.py

actually filter for aqi being None rather than None or 0

Co-authored-by: Erik Montnemery <erik@montnemery.com>

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

astrandb committed:
Add alarm and connectivity binary_sensors to myuplink (#111643)

* Add alarm and connectivity binary_sensors

* Get is_on for correct system

* Make coverage 100% in binary_sensor

* Address review comments

* Revert dict comprehension for now

Bre77 committed:
Add select platform to Teslemetry (#117422)

* Add select platform

* Add tests

* Tests WIP

* Add tests

* Update homeassistant/components/teslemetry/select.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/teslemetry/select.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* use references

* Fix typo

* Update homeassistant/components/teslemetry/select.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/teslemetry/select.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Make less confusing for @joostlek

* Update homeassistant/components/teslemetry/select.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/teslemetry/select.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: G Johansson <goran.johansson@shiftit.se>

mawoka-myblock committed:
Post review comments on APsystems (#117504)

* Cleanup for apsystems and fix for strings

* Migrate to typed ConfigEntry Data for apsystems

* Improve strings for apsystems

* Improve config flow tests for apsystems by cleaning up fixtures

* Do not use Dataclass for Config Entry Typing

* Improve translations for apsystems by using sentence case and removing an apostrophe

* Rename test fixture and remove unnecessary comment in tests from apsystems

* Remove default override with default in coordinator from apsystems

epenet committed:
Move vizio coordinator to separate module (#117498)

epenet committed:
Move vallox coordinator to separate module (#117503)

* Move vallox coordinator to separate module

* Move logic into coordinator class

* Adjust

farmio committed:
Allow templates for enabling conditions (#117047)

* Allow templates for enabling automation conditions

* Use `cv.template` instead of `cv.template_complex`

farmio committed:
Allow templates for enabling actions (#117049)

* Allow templates for enabling automation actions

* Use `cv.template` instead of `cv.template_complex`

* Rename test function

starkillerOG committed:
Add Reolink PIR entities (#117507)

* Add PIR entities

* fix typo

IceBotYT committed:
Add light platform to Linear garage door (#111426)

* Add light platform

* Fix light test

* Suggestions by CFenner

* Fix tests

* More fixes

* Revert test changes

* Undo base entity

* Rebase

* Fix to use base entity

* Fix name

* More fixes

* Fix tests

* Add translation key

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

epenet committed:
Rename fritz coordinator module (#117440)

* Rename fritz coordinator module

* Update .coveragerc

* Adjust .coveragerc

* Adjust coverage

* Adjust coverage

jpbede committed:
Do not register mqtt mock config flow with handlers (#117521)

balloob committed:
Set integration type for wyoming (#117519)

* Set integration type to wyoming

* Add entry_type

balloob committed:
Fix the type of slot schema of intent handlers (#117520)

Fix the slot schema of dynamic intenet handler

Shulyaka committed:
Add LLM tools (#115464)

* Add llm helper

* break out Tool.specification as class members

* Format state output

* Fix intent tests

* Removed auto initialization of intents - let conversation platforms do that

* Handle DynamicServiceIntentHandler.extra_slots

* Add optional description to IntentTool init

* Add device_id and conversation_id parameters

* intent tests

* Add LLM tools tests

* coverage

* add agent_id parameter

* Apply suggestions from code review

* Apply suggestions from code review

* Apply suggestions from code review

* Apply suggestions from code review

* Apply suggestions from code review

* Fix tests

* Fix intent schema

* Allow a Python function to be registered as am LLM tool

* Add IntentHandler.effective_slot_schema

* Ensure IntentHandler.slot_schema to be vol.Schema

* Raise meaningful error on tool not found

* Move this change to a separate PR

* Update todo integration intent

* Remove Tool constructor

* Move IntentTool to intent helper

* Convert custom serializer into class method

* Remove tool_input from FunctionTool auto arguments to avoid recursion

* Remove conversion into Open API format

* Apply suggestions from code review

* Fix tests

* Use HassKey for helpers (see #117012)

* Add support for functions with typed lists, dicts, and sets as type hints

* Remove FunctionTool

* Added API to get registered intents

* Move IntentTool to the llm library

* Return only handlers in intents.async.get

* Removed llm tool registration from intent library

* Removed tool registration

* Add bind_hass back for now

* removed area and floor resolving

* fix test

* Apply suggestions from code review

* Improve coverage

* Fix intent_type type

* Temporary disable HassClimateGetTemperature intent

* Remove bind_hass

* Fix usage of slot schema

* Fix test

* Revert some test changes

* Don't mutate tool_input

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

synesthesiam committed:
Don't prioritize "name" slot if it's a wildcard in default conversation agent (#117518)

* Don't prioritize "name" slot if it's a wildcard

* Fix typing error

bdraco committed:
Bump bleak to 0.22.1 (#117383)

epenet committed:
Move google coordinator to separate module (#117473)

autinerd committed:
Add Webmin filesystem sensors (#112660)

* Add Webmin filesystem sensors

* fix names

* update snapshots

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

epenet committed:
Move xbox coordinator to separate module (#117421)

epenet committed:
Move switcher_kis coordinator to separate module (#117538)

epenet committed:
Move gogogate2 coordinator to separate module (#117433)

gnumpi committed:
Bump aioesphomeapi to 24.4.0 (#117543)

tichris0 committed:
Bump adext to 0.4.3 (#117496)

epenet committed:
Move surepetcare coordinator to separate module (#117544)

epenet committed:
Move upcloud coordinator to separate module (#117536)

thejeffreystone committed:
Add functionality to Mastodon (#112862)

* Adds functionality to Mastodon

* protect media type

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* update log warning

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* protect upload media

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update protected functions

---------

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

autinerd committed:
Only allow ethernet and wi-fi interfaces as unique ID in webmin (#113084)

cdce8p committed:
Improve ReloadServiceHelper typing (#117552)

cdce8p committed:
Replace meaningless TypeVar usage (#117553)

chemelli74 committed:
Add Shelly motion sensor switch (#115312)

* Add Shelly motion sensor switch

* update name

* make motion switch a restore entity

* add test

* apply review comment

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* rename switch

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Update tests/components/shelly/test_switch.py

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* fix ruff

---------

Co-authored-by: Shay Levy <levyshay1@gmail.com>

epenet committed:
Move rainforest_eagle coordinator to separate module (#117556)

epenet committed:
Move risco coordinator to separate module (#117549)

joostlek committed:
Handle uncaught exceptions in Analytics insights (#117558)

dfaour committed:
Improve recorder statistics error messages (#113498)

* Update statistics.py

Added more detail error descriptions to make debugging easier

* Update statistics.py

formatting corrected

epenet committed:
Move pvpc coordinator to separate module (#117559)

joostlek committed:
Fix poolsense naming (#117567)

synesthesiam committed:
Add area filter and rounded time to timers (#117527)

* Add area filter

* Add rounded time to status

* Fix test

* Extend test

* Increase test coverage

epenet committed:
Move pure_energie coordinator to separate module (#117560)

ludeeus committed:
Update hass-nabucasa dependency to version 0.81.0 (#117568)

joostlek committed:
Use runtime data in Poolsense (#117570)

epenet committed:
Ensure gold and platinum integrations implement diagnostic (#117565)

bieniu committed:
Support reconfigure flow in Shelly integration (#117525)

* Support reconfigure flow

* Update strings

* Add tests

---------

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

joostlek committed:
Refactor Poolsense config flow tests (#117573)

ctalkington committed:
Bump pyipp to 0.16.0 (#117583)

bump pyipp to 0.16.0

MatthewFlamm committed:
Add diagnostics for nws (#117587)

* add diagnostics

* remove hassfezt exception

joostlek committed:
Add Poolsense platform tests (#117579)

starkillerOG committed:
Add Reolink battery entities (#117506)

* add battery sensors

* Disable Battery Temperature and State by default

* fix mypy

* Use device class for icon

MatthewFlamm committed:
Use config entry runtime_data in nws (#117593)

cdce8p committed:
Mark HassJob target as Final (#117578)

Kane610 committed:
Rework deCONZ services to load once and never unload (#117592)

* Rework deCONZ services to load once and never unload

* Fix hassfest

Kane610 committed:
Remove old deCONZ entity cleanup (#117590)

teharris1 committed:
Fix issue changing Insteon Hub configuration (#117204)

Add Hub version to config schema

epenet committed:
Rename openweathermap coordinator module (#117609)

epenet committed:
Move opengarage coordinator to separate module (#117608)

bdraco committed:
Fix dlna_dmr task flood when player changes state (#117606)

Kane610 committed:
Improve syncing light states to deCONZ groups (#117588)

dependabot[bot] committed:
Bump actions/checkout from 4.1.4 to 4.1.6 (#117612)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

starkillerOG committed:
Fix Reolink battery translation_key unneeded (#117616)

tr4nt0r commented:
damn, messed up the merge. again. 🙄 
I'm so sorry for all the unwanted notifications.
